### ISSUE TITLE
false-positive on non-private Enum module members

### ISSUE THREAD

**Issue by Infrasonics** [PyCQA/pylint#2804]
### Steps to reproduce
Run pylint -E against this snippet:

```
import signal

print(signal.Signals)
print(signal.Sigmasks)
print(signal._IntEnum)
```
Interestingly the member `_IntEnum` does not raise an error (it's empty).

### Current behavior
```
linterror.py:3:6: E1101: Module 'signal' has no 'Signals' member (no-member)
linterror.py:4:6: E1101: Module 'signal' has no 'Sigmasks' member (no-member)
```
Adding `--extension-pkg-whitelist=signal` does not change the output.

### Expected behavior
No Errors.

### pylint --version output
pylint 2.2.2
astroid 2.1.0
Python 3.6.5 (default, Jun 18 2018, 03:18:01) 
[GCC 6.4.0]

**Comment by brycepg** [comment]
Could you paste the relevant contents of signal package or give the package name and version on pypi?

It appears there's a lot of issues with Enums if you look at the issues history

**Comment by brycepg** [comment]
Looking at the source:

```
from enum import IntEnum as _IntEnum 
_IntEnum._convert(
        'Signals', __name__,
        lambda name:
            name.isupper()
            and (name.startswith('SIG') and not name.startswith('SIG_'))
            or name.startswith('CTRL_'))
```

uses the _convert classmethod:

```
    @classmethod
    def _convert(cls, name, module, filter, source=None):
        """
        Create a new Enum subclass that replaces a collection of global constants
        """
        # convert all constants from source (or module) that pass filter() to
        # a new Enum called name, and export the enum and its members back to
        # module;
        # also, replace the __reduce_ex__ method so unpickling works in
        # previous Python versions
        module_globals = vars(sys.modules[module])
        if source:
            source = vars(source)
        else:
            source = module_globals
        # We use an OrderedDict of sorted source keys so that the
        # _value2member_map is populated in the same order every time
        # for a consistent reverse mapping of number to name when there
        # are multiple names for the same number rather than varying
        # between runs due to hash randomization of the module dictionary.
        members = [
                (name, source[name])
                for name in source.keys()
                if filter(name)]
        try:
            # sort by value
            members.sort(key=lambda t: (t[1], t[0]))
        except TypeError:
            # unless some values aren't comparable, in which case sort by name
            members.sort(key=lambda t: t[0])
        cls = cls(name, members, module=module)
        cls.__reduce_ex__ = _reduce_ex_by_name
        module_globals.update(cls.__members__)
        module_globals[name] = cls
        return cls
```

which modifies module globals. Pylint does not handle this type of code.

I suppose an astroid brain tip for `_IntEnum._convert` would fix this.

**Comment by PCManticore** [comment]
Providing a `signal` brain tip should do it for this particular case, I doubt we can add support for `IntEnum. _convert`.

**Comment by aressler38** [comment]
This is still happening on my system. Is there a workaround we can add locally, or should I disable this pylint error?

```pylint 2.8.2
astroid 2.5.6
Python 3.9.4 (default, Apr 10 2021, 15:31:19) 
[GCC 8.3.0]```

**Comment by hippo91** [comment]
@aressler38 you should disable the error. The other solution would be to code an astroid brain. I would be happy to review it if you want to give it a try.
